tests/bxml.test.js

```diff
@@ -0,0 +1,557 @@
+
+            var gather = new Gather({
+                gatherUrl: "https://test.com",
+                gatherMethod: "GET",
+                username: "user",
+                password: "pass",
+                tag: "tag",
+                terminatingDigits: "123",
+                maxDigits: 3,
+                interDigitTimeout: 4,
+                firstDigitTimeout: 5,
+                audioProducers: [playAudio],
+            });
+
+            var response = new Response(gather);
+
+            var expectedString = "<?xml version=\"1.0\" encoding=\"UTF-8\"?><Response><Gather gatherUrl=\"https://test.com\" gatherMethod=\"GET\" username=\"user\" password=\"pass\" tag=\"tag\" terminatingDigits=\"123\" maxDigits=\"3\" interDigitTimeout=\"4\" firstDigitTimeout=\"5\"><PlayAudio username=\"user\" password=\"pass\">https://test.com</PlayAudio></Gather></Response>";
+            expect(response.toBxml()).toEqual(expectedString);
+        });
+        it("should generate a proper Gather tag with no nested tags", function() {
+            var gather = new Gather({
+                gatherUrl: "https://test.com",
+                gatherMethod: "GET",
+                username: "user",
+                password: "pass",
+                tag: "tag",
+                terminatingDigits: "123",
+                maxDigits: 3,
+                interDigitTimeout: 4,
+                firstDigitTimeout: 5,
+                repeatCount: 6,
+                gatherFallbackUrl: "https://test2.com",
+                gatherFallbackMethod: "POST",
+                fallbackUsername: "fuser",
+                fallbackPassword: "fpass",
+            });
+
+
+            var response = new Response(gather);
+
+            var expectedString = "<?xml version=\"1.0\" encoding=\"UTF-8\"?><Response><Gather gatherUrl=\"https://test.com\" gatherMethod=\"GET\" username=\"user\" password=\"pass\" tag=\"tag\" terminatingDigits=\"123\" maxDigits=\"3\" interDigitTimeout=\"4\" firstDigitTimeout=\"5\" repeatCount=\"6\" gatherFallbackUrl=\"https://test2.com\" gatherFallbackMethod=\"POST\" fallbackUsername=\"fuser\" fallbackPassword=\"fpass\"/></Response>";
+            expect(response.toBxml()).toEqual(expectedString);
+        });
+    });
+});
+
+//Tests for Transfer
+describe("Transfer", function() {
+    describe("#toBxml()", function() {
+        it("should generate a proper Transfer tag", function() {
+            var number1 = new Transfer.PhoneNumber({
+                number: "+17777777777",
+                transferAnswerUrl: "https://test.com",
+                transferAnswerMethod: "GET",
+                username: "user",
+                password: "pass",
+                tag: "tag",
+                transferAnswerFallbackUrl: "https://test2.com",
+                transferAnswerFallbackMethod: "GET",
+                fallbackUsername: "fuser",
+                fallbackPassword: "fpass",
+            });
+
+            var number2 = new Transfer.PhoneNumber({
```

Comment:
We should make this `PhoneNumber` a `SipURI` for better coverage

Reply:
Ooh good catch. Updating...